tests/test_interdiff_command.rs

```diff
@@ -0,0 +1,143 @@
+    index 257cc5642c...3bd1f0e297 100644
+    --- a/file2
+    +++ b/file2
+    @@ -1,1 +1,2 @@
+     foo
+    +bar
+    "###);
+}
+
+#[test]
+fn test_interdiff_paths() {
+    let test_env = TestEnvironment::default();
+    test_env.jj_cmd_success(test_env.env_root(), &["init", "repo", "--git"]);
+    let repo_path = test_env.env_root().join("repo");
+
+    std::fs::write(repo_path.join("file1"), "foo\n").unwrap();
+    std::fs::write(repo_path.join("file2"), "foo\n").unwrap();
+    test_env.jj_cmd_success(&repo_path, &["new"]);
+    std::fs::write(repo_path.join("file1"), "bar\n").unwrap();
+    std::fs::write(repo_path.join("file2"), "bar\n").unwrap();
+    test_env.jj_cmd_success(&repo_path, &["branch", "create", "left"]);
+
+    test_env.jj_cmd_success(&repo_path, &["checkout", "root"]);
+    std::fs::write(repo_path.join("file1"), "foo\n").unwrap();
+    std::fs::write(repo_path.join("file2"), "foo\n").unwrap();
+    test_env.jj_cmd_success(&repo_path, &["new"]);
+    std::fs::write(repo_path.join("file1"), "baz\n").unwrap();
+    std::fs::write(repo_path.join("file2"), "baz\n").unwrap();
+    test_env.jj_cmd_success(&repo_path, &["branch", "create", "right"]);
+
+    let stdout = test_env.jj_cmd_success(
+        &repo_path,
+        &["interdiff", "--from", "left", "--to", "right", "file1"],
+    );
+    insta::assert_snapshot!(stdout, @r###"
+    Modified regular file file1:
+       1    1: barbaz
+    "###);
+
+    let stdout = test_env.jj_cmd_success(
+        &repo_path,
+        &["interdiff", "--from", "left", "--to", "right", "file1", "file2"],
+    );
+    insta::assert_snapshot!(stdout, @r###"
+    Modified regular file file1:
+       1    1: barbaz
+    Modified regular file file2:
+       1    1: barbaz
+    "###);
+}
+
+#[test]
+fn test_interdiff_conflicting() {
+    let test_env = TestEnvironment::default();
+    test_env.jj_cmd_success(test_env.env_root(), &["init", "repo", "--git"]);
+    let repo_path = test_env.env_root().join("repo");
+
+    std::fs::write(repo_path.join("file"), "foo\n").unwrap();
+    test_env.jj_cmd_success(&repo_path, &["new"]);
+    std::fs::write(repo_path.join("file"), "bar\n").unwrap();
+    test_env.jj_cmd_success(&repo_path, &["branch", "create", "left"]);
+
+    test_env.jj_cmd_success(&repo_path, &["checkout", "root"]);
+    std::fs::write(repo_path.join("file"), "abc\n").unwrap();
+    test_env.jj_cmd_success(&repo_path, &["new"]);
+    std::fs::write(repo_path.join("file"), "def\n").unwrap();
+    test_env.jj_cmd_success(&repo_path, &["branch", "create", "right"]);
+
+    let stdout = test_env.jj_cmd_success(
+        &repo_path,
+        &["interdiff", "--from", "left", "--to", "right"],
+    );
+    insta::assert_snapshot!(stdout, @r###"
+    Resolved conflict in file:
+       1    1: <<<<<<<def
+       2     : -------
+       3     : +++++++
+       4     : -foo
+       5     : +abc
+       6     : +++++++
+       7     : bar
+       8     : >>>>>>>
+    "###);
```

Comment:
This is what the output _is_, but this is completely opaque to me and I'm not sure what the expected output _should_ be.

Reply:
It's probably easier to read with `--git` since that format doesn't rely on color.

Reply:
Here's an example [from Discord](https://discord.com/channels/968932220549103686/969829516539228222/982408018270511164):

<blockquote>
so even when we support regular 3-way markers, we'll probably want to fall back to this format
Instead of:

```
<<<<<<<
left
=======
base
|||||||
right
>>>>>>>
```

These markers are:

```
<<<<<<<
-------
+++++++
-base
+left
+++++++
right
>>>>>>>
```

So what you'll want to do is to look at the diff part and apply it to the snapshot part. Does that make sense?
I.e. look at the diff between base and left and apply it to right part in this example
</blockquote>

@martinvonz is that officially documented anywhere? I didn't see it in `conflicts.md`.

Reply:
> @martinvonz is that officially documented anywhere? I didn't see it in `conflicts.md`.

Good question. I don't think I've documented that anywhere. I'll try to remember to write about the format soon.

Reply:
By the way, when displaying a diff involving conflicts like this, one *could* display it kind of like with git's `--cc`:

```
- bar
+ foo
 -abc
 +def
```

But that's not particularly easy to read either :) When a merge commit resolves a conflict by picking the content from one side of the merge, that format would show you a diff saying that the other side of the merge was discarded.

I think part of the reason these look opaque is that it's a four-way diff of an artificial diff with unrelated contents. It might not be easy to understand on some of your real commits, but it's hopefully at least much better there. (I haven't tried it myself.)

Reply:
`--git` is better, yeah:

```
    -<<<<<<<
    --------
    -+++++++
    --foo
    -+abc
    -+++++++
    -bar
    ->>>>>>>
    +def
```

i.e., right/`+` has `def` and left/`-` has a conflict.